dapp/src/queries/useBalancesQuery.js

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from 'react-query'
+
+import { QUERY_KEYS } from 'constants/queryKeys'
+
+import { balancesService } from '../services/balances.service'
+
+const useBalancesQuery = (account, contracts, options) => {
+  return useQuery(
+    QUERY_KEYS.Balances(account),
+    () => balancesService.fetchBalances(account, contracts),
+    {
+      enabled: account != null && contracts != null,
+      refetchOnWindowFocus: false,
```

Comment:
The `enabled` flag allows the query to only run when we got an account and the contracts ready.

I've disabled the `refetchOnWindowFocus` flag for now because it's a bit too aggressive. In the future we can use blocknative to detect if a user balances has changed and trigger a refetch on demand.

Other logic could be to refetch on each block etc.

Reply:
Nice, can you subscribe to Transfer and Approaval events of an account? if yes that would be a nice performance and UX improvement and should also simplify code to some extent.